src/Operation/InsertMany.php

```diff
@@ -142,6 +153,15 @@ public function execute(Server $server)
         $insertedIds = [];
 
         foreach ($this->documents as $i => $document) {
+            if (isset($this->options['codec'])) {
+                $document = $this->options['codec']->encodeIfSupported($document);
+            }
+
+            // Psalm's assert-if-true annotation does not work with unions, so
+            // assert the type manually instead of using is_document
+            // See https://github.com/vimeo/psalm/issues/6831
+            assert(is_array($document) || is_object($document));
```

Comment:
I think this assertion can be moved within `if (isset($this->options['codec']))` above.